tools/cloud_storage_manifest_patcher/main.py

```diff
@@ -0,0 +1,393 @@
+import boto3
+from botocore.config import Config
+from botocore.exceptions import ClientError
+from time import sleep
+from functools import wraps
+from collections import namedtuple
+import time
+import datetime
+from typing import Union
+import json
+
+
+class SlowDown(Exception):
+    pass
+
+
+S3ObjectMetadata = namedtuple('S3ObjectMetadata',
+                              ['Bucket', 'Key', 'ETag', 'ContentLength'])
+
+
+def retry_on_slowdown(tries=4, delay=1.0, backoff=2.0):
+    """Retry with an exponential backoff if SlowDown exception was triggered."""
+    def retry(fn):
+        @wraps(fn)
+        def do_retry(*args, **kwargs):
+            ntries, sec_delay = tries, delay
+            while ntries > 1:
+                try:
+                    return fn(*args, **kwargs)
+                except SlowDown:
+                    sleep(sec_delay)
+                    ntries -= 1
+                    sec_delay *= backoff
+            # will stop filtering SlowDown exception when quota is reache
+            return fn(*args, **kwargs)
+
+        return do_retry
+
+    return retry
+
+
+class S3Client:
+    """Simple S3 client"""
+    def __init__(self,
+                 region,
+                 access_key,
+                 secret_key,
+                 logger,
+                 endpoint=None,
+                 disable_ssl=True):
+        cfg = Config(region_name=region, signature_version='s3v4')
+        self._cli = boto3.client('s3',
+                                 config=cfg,
+                                 aws_access_key_id=access_key,
+                                 aws_secret_access_key=secret_key,
+                                 endpoint_url=endpoint,
+                                 use_ssl=not disable_ssl)
+        self._region = region
+        self.logger = logger
+
+    def create_bucket(self, name):
```

Comment:
Thank you for adding some type annotations. Ideally we'd have all function types (args and return values) annotated as outlined in our (evolving) python standards (see wiki page "Python Standardization" for details. I'm happy to help add annotations later (as followup PR) if that helps.